pkg/controller/operators/olm/operator.go

```diff
@@ -317,6 +318,35 @@ func (a *Operator) syncObject(obj interface{}) (syncError error) {
 		a.requeueOwnerCSVs(metaObj)
 	}
 
+	// Requeues objects that can't have ownerrefs (cluster -> namespace, cross-namespace)
+	if ownerutil.IsOwnedByKindLabel(metaObj, v1alpha1.ClusterServiceVersionKind) {
+		logger.Debug("requeueing owner CSVs")
+		a.requeueOwnerCSVs(metaObj)
+	}
+
+	// TODO: only check this on namespace add/delete, not on every namespace sync
+	// Check to see if any operator groups are associated with this namespace
+	namespace, ok := obj.(*corev1.Namespace)
+	if !ok {
+		return nil
+	}
+
+	operatorGroupList, err := a.lister.OperatorsV1alpha2().OperatorGroupLister().OperatorGroups(metav1.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		syncError = fmt.Errorf("lister failed: %v", err)
+		logger.Warn(syncError.Error())
+		return
+	}
+
+	for _, group := range operatorGroupList {
+		_, ok := resolver.NewNamespaceSet(group.Status.Namespaces).Intersection(resolver.NewNamespaceSet([]string{namespace.GetName()}))[namespace.GetName()]
```

Comment:
Why generate a new `NamespaceSet` for `namespace` on every iteration of the loop? You should only need to generate a new one each iteration for `group.Status.Namespaces`. Also, rather than checking for `namespace.GetName`'s existence in the set, you can just check if `|intersection| > 0`.

Reply:
Using the `Contains` method is probably better: https://github.com/operator-framework/operator-lifecycle-manager/blob/e1e34a0e3acc16d9c48ff31b5f2d44aea0f6dfbb/pkg/controller/registry/resolver/groups.go#L52